java-manta-client/src/main/java/com/joyent/manta/client/multipart/EncryptionStateSnapshot.java

```diff
@@ -83,8 +106,8 @@ public boolean equals(final Object o) {
         }
 
         final EncryptionStateSnapshot that = (EncryptionStateSnapshot) o;
-        return lastPartNumber == that.lastPartNumber
-                && cipher == that.cipher
-                && hmac == that.hmac;
+        return Objects.equals(uploadId, that.uploadId)
```

Comment:
They're not strictly necessary but it was a matter of habit. The fact that `EncryptionState` has those methods was the primary motivation.